superset/explore/form_data/commands/create.py

```diff
@@ -40,7 +41,10 @@ def run(self) -> str:
             actor = self._cmd_params.actor
             form_data = self._cmd_params.form_data
             check_access(dataset_id, chart_id, actor)
-            key = token_urlsafe(48)
+            contextual_key = cache_key(actor.get_user_id(), dataset_id, chart_id)
```

Comment:
1. In order to separate different session for same user, use session id instead of user id
```python
from flask import session
...
...
            contextual_key = cache_key(session.get('_id'), dataset_id, chart_id)
```

2. there is already a general hash module in `superset.utils.hashing`
```
import from md5_sha_from_str from superset.utils.hashing

contextual_key = md5_sha_from_str(<string variable>)
```

Reply:
1. Great call on the different sessions approach! I modified the code to follow your suggestion.
2. I think we don't need the extra overhead of a hashing function. The concatenation is just fine for this use case. Even if we would apply a hashing function we would still need to concatenate the arguments. I left this untouched. Let me know if you disagree 😉

Reply:
The objective is the same: get a unique cache. Thanks for the improvement.